### PR TITLE
EOS-26013: Performance Stats: Create perfstat topic and configure it for topic level persistence in mini-provisioining phase

### PR DESCRIPTION
**Pull request by udayan-y21**
# Problem Statement
- [EOS-26029](https://jts.seagate.com/browse/EOS-26029)

# Design
- Prepare phase: Fetch retention period and message type from confstore and set in csm config
- Config Phase: Create message type in message bus using MessageBusAdmin interface
- Expose POST API: Consume MessageBusComm class and initialize producer. 
- Publish messages to messagebus 

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
Miniprovisioning Testing
image

Generated csm.conf  snipet gathering messagebus related config
image

POST Api for publishing messages:
image

Test using Consumer:
```
>>> import sys, os
>>> sys.path.append('/opt/seagate/cortx')
>>> from csm.common.comm import MessageBusComm
>>> message_bus = MessageBusComm()
>>> message_bus.init(type='consumer', consumer_id='csm',consumer_group='test_group', consumer_message_types=["perf_stat"],auto_ack=False, offset="earliest")
>>> def callback_fn(message):
...     print(f"Message received: {message}")
...
>>> message_bus.recv(callback_fn)
Message received: message-1
Message received: message-2
Message received: message-3
```
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: csm/conf/prepare.py [hunk 1]
```diff
@@ -115,7 +116,9 @@ def _prepare_and_validate_confstore_keys(self):
                 const.OPENLDAP_SERVERS:f"{const.OPENLDAP_SERVERS_KEY}",
                 const.OPENLDAP_ROOT_ADMIN:f"{const.OPENLDAP_ROOT_ADMIN_KEY}",
                 const.OPENLDAP_ROOT_SECRET:f"{const.OPENLDAP_ROOT_SECRET_KEY}",
-                const.OPENLDAP_BASEDN:f"{const.OPENLDAP_BASEDN_KEY}"
+                const.OPENLDAP_BASEDN:f"{const.OPENLDAP_BASEDN_KEY}",
+                const.METRICS_PERF_STATS_MSG_TYPE : const.METRICS_PERF_STATS_MSG_TYPE_KEY,
+                const.METRICS_PERF_STATS_RETENTION_PERIOD:const.METRICS_PERF_STATS_RETENTION_PERIOD_KEY
```
Comment (pawankumarsrivastava) [inline csm/conf/prepare.py:121]: Change retention period to retention size

File: csm/conf/configure.py [hunk 2]
```diff
@@ -487,3 +489,14 @@ def _set_user_collection(self):
                 record['config']['openldap']['collection'] = const.CORTXUSERS_DN.format(csm_schema_version,base_dn)
         Conf.set(const.DATABASE_INDEX,"models",models_list)
         Conf.save(const.DATABASE_INDEX)
+
+    def _create_perf_stat_topic(self):
+        mb_admin = MessageBusAdmin(admin_id = Conf.get(const.CSM_GLOBAL_INDEX,const.MSG_BUS_ADMIN_ID))
+        message_type = Conf.get(const.CSM_GLOBAL_INDEX,const.MSG_BUS_PERF_STAT_MSG_TYPE)
+        partitions = Conf.get(const.CSM_GLOBAL_INDEX,const.MSG_BUS_PERF_STAT_PARTITIONS)
+        retention_period = Conf.get(const.CSM_GLOBAL_INDEX,const.MSG_BUS_PERF_STAT_RETENTION_PERIOD)
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:497]: Change retention_period to retention_size

File: csm/core/agent/csm_agent.py [hunk 3]
```diff
@@ -122,10 +123,14 @@ def init():
        # Network file manager registration
         CsmRestApi._app["download_service"] = DownloadFileManager()
 
+        #Message Bus instance creation
+        CsmRestApi._app["comm_client"] = MessageBusComm()
```
Comment (pawankumarsrivastava) [inline csm/core/agent/csm_agent.py:127]: No need for this. You need to create a function in stats service call stop. In that function call comm client's stop.

File: csm/core/agent/csm_agent.py [hunk 4]
```diff
@@ -261,6 +266,8 @@ def run():
         if not (env_type == const.K8S):
             CsmAgent.alert_monitor.stop()
             Log.info("Finished stopping alert monitor service")
+        Log.info("Stopping Message Bus client")
+        CsmRestApi._app["comm_client"].stop()
```
Comment (pawankumarsrivastava) [inline csm/core/agent/csm_agent.py:270]: Call CsmRestApi._app["stat_service"].stop() instead of CsmRestApi._app["comm_client"].stop()

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/conf/prepare.py (hunk 1 above)
Comment (udayan-y21) [reply csm/conf/prepare.py]: Modifications done

File: csm/conf/configure.py (hunk 2 above)
Comment (udayan-y21) [reply csm/conf/configure.py]: Modifications done

File: csm/core/agent/csm_agent.py (hunk 3 above)
Comment (udayan-y21) [reply csm/core/agent/csm_agent.py]: Modifications done

File: csm/core/agent/csm_agent.py (hunk 4 above)
Comment (udayan-y21) [reply csm/core/agent/csm_agent.py]: Modifications done

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by Pranali04796 — APPROVED** [review]

**Pull request merged**